Added Reason option for remove/add roles & fixed $interactionDeferUpdate

ethann2004 opened the pull request:
## Type
- [ x] Bug Fix
- [ x] Functions: `$giveRoles`, `$giveRole`, `$takeRoles`, `$takeRole`, `$interactionDeferUpdate`, `$authorAvatar`, `$memberAvatar`, `$userAvatar`, `$userDefaultAvatar`

## Description
I added the `Reason` option for the give/take role functions. I also fixed `$interactionDeferUpdate` return [Object, Object]. Changed `format` to `extension` in all avatar functions since Discord.js v14 renamed that option.

ethann2004 committed:
Added "reason" option

ethann2004 committed:
Added "reason" option

ethann2004 committed:
Added "reason" option

ethann2004 committed:
Added "reason" option

ethann2004 committed:
Fixed the issue where it'll return [Object, Object]

ethann2004 committed:
Format was changed to extension in discord.js v14

ethann2004 committed:
Format was changed to extension in discord.js v14

ethann2004 committed:
Format was changed to extension in discord.js v14

ethann2004 committed:
Format was changed to extension in discord.js v14

ethann2004 committed:
Format was changed to extension in discord.js v14

Neodevils commented:
Hello @ethann2004! 

I would like to recommend you to **create a pull** request for per file/thing. It will be easier to manage pull requests.

Leref commented:
> Hello @ethann2004!
> 
> I would like to recommend you to **create a pull** request for per file/thing. It will be easier to manage pull requests.

It wasn’t hard to read since the whole purpose of this PR was to add `reason` of functions that have support for it.

USERSATOSHI commented on src/functions/Funcs/interaction/giveRole.js:
```diff
@@ -2,7 +2,7 @@ module.exports = async d => {
     if (data.err) return d.error(data.err);
 
-    const [guildID, userID, roleId] = data.inside.splits;
+    const [guildID, userID, reason, roleId] = data.inside.splits;
```
i think reason should be the last field as it can be optional

USERSATOSHI commented on src/functions/Funcs/interaction/takeRole.js:
```diff
@@ -2,19 +2,19 @@ module.exports = async d => {
     if (data.err) return d.error(data.err);
 
-    const [guildID, userID, roleId] = data.inside.splits;
+    const [guildID, userID, reason, roleId] = data.inside.splits;
```
same here

USERSATOSHI commented on src/functions/Funcs/user/userAvatar.js:
```diff
@@ -1,18 +1,18 @@
 
     data.result = user.displayAvatarURL({
         size: Number(size),
         dynamic: dynamic === 'true',
```
this can be changed to forceStatic:  dynamic === 'false'

USERSATOSHI commented on src/functions/Funcs/user/userDefaultAvatar.js:
```diff
@@ -1,7 +1,7 @@
     const data = d.util.aoiFunc(d);
 
-    const [userID = d.author?.id, size = 4096, dynamic = 'true', format] = data.inside.splits;
+    const [userID = d.author?.id, size = 4096, dynamic = 'true', extension] = data.inside.splits;
```
defaultAvatarUrl doesnt support properties so we can remove image options and just keep userid

ethann2004 commented:
@USERSATOSHI I'll edit the files later on when I have time to spare!

USERSATOSHI commented:
> @USERSATOSHI I'll edit the files later on when I have time to spare!

Sure np

ethann2004 committed:
Changed parameters

ethann2004 committed:
Changed parameters

ethann2004 committed:
Changed dynamic to forceStatic

ethann2004 committed:
Changed dynamic to forceStatic

ethann2004 committed:
Changed dynamic to forceStatic

ethann2004 committed:
Removed size, dynamic, and extension options

ethann2004 committed:
Added "reason" option

ethann2004 commented:
@USERSATOSHI I think I fixed everything you mention! Let me know if you find something else I can change. I also added the "reason" option in `$setRoles` as I forgot
